assets/product-info.js

```diff
@@ -157,12 +153,20 @@ if (!customElements.get('product-info')) {
         if (variantSelects) this.variantSelectors.innerHTML = variantSelects.innerHTML;
       }
 
-      handleUpdateProductInfo(baseUrl, variant) {
+      handleUpdateProductInfo(baseUrl) {
         return (html) => {
+          const variant = this.getSelectedVariant(html);
+
           this.pickupAvailability?.update(variant);
-          this.updateMedia(html, variant?.featured_media?.id);
           this.updateOptionValues(html);
           this.updateURL(baseUrl, variant?.id);
+          this.updateMedia(html, variant?.featured_media?.id);
+          this.updateVariantInputs(variant?.id);
```

Comment:
Note that these 2 lines are a meaningful behavioral change when switching to an option value combination that has no associated variant!

The impact here is that:
1) We clear any previous variant's featured media, the same as we do when switching to a variant without featured media.
2) We clear the variant ID value in the product form

Reply:
I didn't realize before but I think some of the updateMedia changes (done recently) create an issue where if I have a variant with a media attached and other variants with no media attached, it gives me nothing as the main media shown: [recording](https://screenshot.click/04-31-3j6ex-u2clb.mp4)

Reply:
PR with that fix here: Shopify/dawn/pull/3505